src/cow-react/common/services/getQuoteCurrency/index.ts

```diff
@@ -41,11 +44,13 @@ export function getQuoteCurrency(
 }
 
 export function getQuoteCurrencyByStableCoin(
-  chainId: SupportedChainId | undefined,
+  _chainId: SupportedChainId | undefined,
   inputCurrency: Currency | null,
   outputCurrency: Currency | null
 ): Currency | null {
-  if (!chainId || !inputCurrency || !outputCurrency) return null
+  const chainId = supportedChainId(_chainId)
+
+  if (!chainId || !supportedChainId(chainId) || !inputCurrency || !outputCurrency) return null
```

Comment:
Do you need to check both? Also, you are using the return of `supportedChainId()` and running through the same fn again

 ```suggestion
  if (!chainId || !inputCurrency || !outputCurrency) return null
```